Marlin/Marlin_main.cpp

```diff
@@ -3767,6 +3776,54 @@ inline void gcode_M206() {
 
 #endif // FWRETRACT
 
+#ifdef ENABLE_M211
+/**
+ * M211: Enable or disable software endstops
+ */
+inline void gcode_M211() {
+  // manage bit mask in enable_software_endstops
+  uint8_t smask = 0x00;
+
+  if (code_seen_str("X0")) smask |= 1;
+  if (code_seen_str("Y0")) smask |= 2;
+  if (code_seen_str("Z0")) smask |= 4;
+  if (code_seen_str("X1")) smask |= 16;
+  if (code_seen_str("Y1")) smask |= 32;
+  if (code_seen_str("Z1")) smask |= 64;
```

Comment:
Normally this would be something along the lines of:
```
if (code_seen('X')) smask |= code_value_long() ? 0x10 : 0x01;
if (code_seen('Y')) smask |= code_value_long() ? 0x20 : 0x02;
if (code_seen('Z')) smask |= code_value_long() ? 0x40 : 0x04;
```
...which should run faster, though it might take more SRAM. I would compare. But, your code will allow `M211 X0 X1`. Is that a legal input?